repo2docker/app.py

```diff
@@ -53,10 +54,10 @@ def _default_log_level(self):
         return logging.INFO
 
     git_workdir = Unicode(
-        "/tmp",
+        tempfile.mkdtemp(prefix='repo2docker'),
```

Comment:
we're creating this unconditionally, even if the repo is just already locally existing. I think this should be made conditional

Reply:
True that. We could use `tempfile.gettempdir()` to get the tmp directory but then we still have to somehow generate a directory inside of it. I'd prefer using something from the stdlib instead of doing our own thing (a bit "never roll your own crypto"). However there is no way (I think) of reserving a temporary filename without also creating it.

👍 on this changing for #101 this PR mainly started as a way to get my feet wet 🌊 ...creating a temporary name in a secure way will still be part of that though.

Reply:
How about making the default be 'None' and using tempfile when git_workdir is None?